src/main/kotlin/io/github/anki/anki/controller/CardsController.kt

```diff
@@ -64,42 +64,65 @@ class CardsController(
-            deckId = deckId,
-            userId = securityService.jwtUtils.getUserIdFromAuthHeader(header),
-            pagination = PaginationDto(limit, offset).toPagination(),
-        ).map { it.toDto() }
+    ): Mono<List<CardDtoResponse>> =
+        securityService
+            .getUserIdFromAuthentication()
+            .doFirst { LOG.info("IN: $CardsController $BASE_URL get all cards in deck with id = $deckId") }
+            .flatMapMany {
+                cardService.findCardsByDeckWithPagination(
+                    deckId = deckId,
+                    userId = it,
+                    pagination = PaginationDto(limit, offset).toPagination(),
+                )
+            }
+            .map(Card::toDto)
+            .collectList()
+            .doOnNext { LOG.info("OUT: $CardsController $BASE_URL got cards $it from deck with id = $deckId") }
 
     @PatchMapping(CONCRETE_CARD)
     @ResponseStatus(HttpStatus.OK)
     fun patchCard(
         @PathVariable deckId: String,
         @PathVariable cardId: String,
-        @RequestHeader header: HttpHeaders,
         @RequestBody request: PatchCardRequest,
-    ): CardDtoResponse {
-        LOG.info("IN: $CardsController ${BASE_URL} patch card with id $cardId from deck with id = $deckId")
-        val card =
-            cardService.updateCard(
-                userId = securityService.jwtUtils.getUserIdFromAuthHeader(header),
-                request.toCard(cardId, deckId),
-            )
-        LOG.info("OUT: $CardsController ${BASE_URL} patched card with id $cardId from deck with id = $deckId")
-        return card.toDto()
-    }
+    ): Mono<CardDtoResponse> =
+        securityService
+            .getUserIdFromAuthentication()
+            .doFirst {
+                LOG.info("IN: $CardsController $BASE_URL patch card with id $cardId from deck with id = $deckId")
+            }
+            .flatMap {
+                cardService.updateCard(
+                    userId = it,
+                    request.toCard(cardId, deckId),
+                )
+            }
+            .map(Card::toDto)
+            .doOnNext {
+                LOG.info("OUT: $CardsController $BASE_URL patched card with id $cardId from deck with id = $deckId")
+            }
 
     @DeleteMapping(CONCRETE_CARD)
     @ResponseStatus(HttpStatus.NO_CONTENT)
-    fun deleteCard(@PathVariable deckId: String, @PathVariable cardId: String, @RequestHeader header: HttpHeaders) {
-        LOG.info("IN: $CardsController ${BASE_URL} delete card with id $cardId from deck with id = $deckId")
-        cardService.deleteCard(
-            deckId = deckId,
-            userId = securityService.jwtUtils.getUserIdFromAuthHeader(header),
-            cardId = cardId,
-        )
-        LOG.info("OUT: $CardsController ${BASE_URL} patched card with id $cardId from deck with id = $deckId")
-    }
+    fun deleteCard(
+        @PathVariable deckId: String,
+        @PathVariable cardId: String,
+    ): Mono<Void> =
```

Comment:
Why `Void`? Can we use `Unit`?

Reply:
yeas we can, i will change return type of repository method